### PR TITLE
Fixes #660 by adding windows specific extension

### LINKED ISSUE #660: hie-wrapper script not working on windows
The rewritten hie-wrapper script doesn't work on windows. This is because executable files in Windows have the extension .exe, and thus `findExecutable` fails to find it when looking for `hie-8.2.2`. On windows it should probably add the `.exe` part when searching.

### PR DESCRIPTION
**Pull request by Supernerd11**
This pull request fixes #660 by adding the required extension on windows.

### CODE REVIEW

**Comment by Bubba** [comment]
Should we also add this check for`getProjectGhcVersion` in Base.hs?

**Comment by Supernerd11** [comment]
That would probably be a good idea. Basically anywhere `findExecutable` is used.

**Comment by Supernerd11** [comment]
@Bubba We don't actually. findExecutable seems to find an executable when used without extension. The bug was that it read the version number as it's extension, eg `hie-8.2.2` would assume that `.2` was the extension.

File: app/HieWrapper.hs [hunk 1]
```diff
@@ -90,9 +90,9 @@ run opts = do
       args <- getArgs
       logm $ "args:" ++ show args
       logm "launching ....\n\n\n"
-      callProcess e args
+      callProcess e ("--lsp" : args)
```
Comment (Bubba) [inline app/HieWrapper.hs]: Not sure if we should do this by default, there does exist a non-lsp JSON mode that should be able to use the wrapper too

Comment (Supernerd11) [reply app/HieWrapper.hs]: This was mostly because of the way both the vscode and atom plugins handled custom wrappers. It could probably be removed.

Comment (Bubba) [reply app/HieWrapper.hs]: Looks like https://github.com/alanz/vscode-hie-server/pull/93 will add it for us, but its not merged yet

Comment (Supernerd11) [reply app/HieWrapper.hs]: I know. A better workaround for now would be to make a local batch/shell script to launch the wrapper with the lsp flag.

Comment (Supernerd11) [reply app/HieWrapper.hs]: Or even easier, change default executable to hie-wrapper.

**Pull request merged**